main.rb

```diff
@@ -0,0 +1,114 @@
+  distance_from_sun_km = gets.chomp
+  puts "Pls tell me a fun fact about planet #{name}:"
+  fun_fact = gets.chomp.capitalize
+
+  new_planet = Planet.new(name, color, mass_kg, distance_from_sun_km, fun_fact)
+
+  return new_planet
+end
+
+
+def main
+
+  mercury = Planet.new('Mercury', 'Dark Gray', '328500000000000000000000', '57910000', 'Several times per year, from the perspective of Earth, Mercury appears to be moving backward through the sky; this is the sole reason that things go wrong in my life')
+
+  venus = Planet.new('Venus', 'Pale Yellow', '4867000000000000000000000', '108200000', 'Venus is the name of the women\'s razors division of The Gillette Company')
+
+  earth = Planet.new('Earth', 'Blue', '5972000000000000000000000', '149600000', 'Overall a total shithole, but occasionally produces decent works of art; inhabitants seem generally unaware that rest of universe exists')
+
+  mars = Planet.new('Mars', 'Red', '639000000000000000000000', '227900000', 'Some scientists postulate that humanity actually originated on Mars and was clandestinely transported to Earth with our memories wiped after we destroyed the entire planet in a nuclear holocaust')
+
+  jupiter = Planet.new('Jupiter', 'Orange and White', '1898000000000000000000000000', '778500000', 'Has a terrifying red eyeball eternally staring down the rest of the solar system')
+
+  saturn = Planet.new('Saturn', 'Pale Gold', '568300000000000000000000000', '1433000000', 'Saturn\'s rings are home to an Intergalactic Council that secretly governs all affairs in our solar system')
+
+  uranus = Planet.new('Uranus', 'Pale Blue', '86810000000000000000000000', '2871000000', 'Uranus was officially discovered by Sir William Herschel in 1781')
+
+  neptune = Planet.new('Neptune', 'Pale Blue', '102400000000000000000000000', '4495000000', 'Neptune\'s Spatula was episode 19b of season 1 of the acclaimed children\'s television show SpongeBob Squarepants')
+
+  pluto = Planet.new('Pluto', 'Light Brown', '13100000000000000000000', '5906292480', 'Not actually considered a full-fledged planet at this time; presently downgraded to "Dwarf Planet" status')
+
+  solar_system = SolarSystem.new('Sol')
+  solar_system.add_planet(mercury)
+  solar_system.add_planet(venus)
+  solar_system.add_planet(earth)
+  solar_system.add_planet(mars)
+  solar_system.add_planet(jupiter)
+  solar_system.add_planet(saturn)
+  solar_system.add_planet(uranus)
+  solar_system.add_planet(neptune)
+  solar_system.add_planet(pluto)
+
+
+
+
+  keep_going = true
+
+  while keep_going == true
+    puts "What would you like to do? options are: LIST PLANETS, PLANET DETAILS, ADD PLANET, DISTANCE and EXIT"
+    user_input = gets.chomp
+
+    case user_input.upcase
+    when "EXIT"
+      keep_going = false
+
+    when "LIST PLANETS"
+      puts solar_system.list_planets
+
+    when "PLANET DETAILS"
+      puts "\nWhich planet would you like to know more about?"
+      planet = gets.chomp.downcase
+
```

Comment:
This `when` clause is just enough work that it might make sense to make it a separate method.